javascript/ql/src/semmle/javascript/frameworks/Redux.qll

```diff
@@ -0,0 +1,1285 @@
+       */
+      DataFlow::SourceNode getAComponentTransformer() {
+        result = this
+        or
+        exists(FunctionCompositionCall compose |
+          getAComponentTransformer().flowsTo(compose.getAnOperandNode()) and
+          result = compose
+        )
+      }
+
+      /**
+       * Gets a data-flow node that should flow to `props.name` via the `mapDispatchToProps` function.
+       */
+      DataFlow::Node getDispatchPropNode(string name) {
+        // Implicitly bound by bindActionCreators:
+        //
+        //   const mapDispatchToProps = { foo }
+        //
+        result = getMapDispatchToProps().getMember(name).getARhs()
+        or
+        //
+        //   const mapDispatchToProps = dispatch => ( { foo } )
+        //
+        result = getMapDispatchToProps().getReturn().getMember(name).getARhs()
+        or
+        // Explicitly bound by bindActionCreators:
+        //
+        //   const mapDispatchToProps = dispatch => bindActionCreators({ foo }, dispatch);
+        //
+        exists(BindActionCreatorsCall bind |
+          bind.flowsTo(getMapDispatchToProps().getReturn().getARhs()) and
+          result = bind.getOptionArgument(0, name)
+        )
+      }
+
+      /**
+       * Gets the React component decorated by this call, if one can be determined.
+       */
+      ReactComponent getReactComponent() {
+        exists(DataFlow::SourceNode component | component = result.getAComponentCreatorReference() |
+          component.flowsTo(getAComponentTransformer().getACall().getArgument(0))
+          or
+          component.(DataFlow::ClassNode).getADecorator() = getAComponentTransformer()
+        )
+      }
+    }
+
+    /** A call to `connect`. */
+    private class RealConnectFunction extends ConnectCall {
+      RealConnectFunction() {
+        this = API::moduleImport("react-redux").getMember("connect").getACall()
+      }
+
+      override API::Node getMapStateToProps() { result = getParameter(0) }
+
+      override API::Node getMapDispatchToProps() { result = getParameter(1) }
+    }
+
+    /**
+     * An entry point in the API graphs corresponding to functions named `mapDispatchToProps`,
+     * used to catch cases where the call to `connect` was not found (usually because of it being
+     * wrapped in another function, which API graphs won't look through).
+     */
+    private class HeuristicConnectEntryPoint extends API::EntryPoint {
+      HeuristicConnectEntryPoint() { this = "react-redux-connect" }
+
+      override DataFlow::Node getARhs() { none() }
+
+      override DataFlow::SourceNode getAUse() {
+        exists(DataFlow::CallNode call |
+          call.getAnArgument().asExpr().(Identifier).getName() =
+            ["mapStateToProps", "mapDispatchToProps"] and
+          // exclude genuine calls to avoid duplication
+          not call = DataFlow::moduleMember("react-redux", "connect").getACall() and
+          result = call.getCalleeNode().getALocalSource()
+        )
+      }
+    }
```

Comment:
This qldoc needs a rewrite. 
The result does not correspond to functions named `mapDispatchToProps`, it corresponds to function that take a `mapStateToProps`/`mapDispatchToProps`. 

Maybe: 
```Text
An API entry point corresponding to a `connect` function. 
The `connect` call is recognized based on an argument being named either `mapStateToProps` or `mapDispatchToProps`. 
Used to catch cases where the `connect` function was not recognized by API graphs (usually because of it being
wrapped in another function, which API graphs won't look through). 
```

I also don't see any tests of this EntryPoint.